tracing/src/main/java/com/palantir/tracing/DeferredTracer.java

```diff
@@ -39,9 +40,19 @@
  */
 public final class DeferredTracer {
     private final Optional<Trace> trace;
+    private final Optional<String> operation;
 
     public DeferredTracer() {
+        this(Optional.empty());
+    }
+
+    public DeferredTracer(@CompileTimeConstant String operation) {
```

Comment:
Please remove CompileTimeConstant

Reply:
(because it's reasonable for this to not be a compiletimeconstant, and it is overly onerous for downstream developers)